### PR TITLE
Fixes Damage between PvE and PvP interactions 

### PR DESCRIPTION
**Pull request by Lynx22330**
## About The Pull Request
<details>
<summary>
	Fixes weapon damage, both melee, and projectile based. Mobs now respect armor calculations as well. 
	
	Many thanks to Trilby for helping this all work, and many thanks to Sigmasquad as well for investing as much time as I have. 
</summary>
<hr>

	
<hr>
</details>

## Changelog
:cl:

/:cl:

### CODE REVIEW

**Comment by Lynx22330** [comment]
If you want to test this locally, please do. 

These are the damage calculations.

Melee : 

`Total Damage - (Armor / Armor Divisor) = Result Damage`

Ranged :

`((Base Projectile Damage * Weapon Projectile Multiplier) - ( Armor / Armor Divisor )) * Wound Scale = Result Damage`

**Comment by Lynx22330** [comment]
Also commenting out  a moved = 1 variable in superior_animal.dm so mobs do 100% damage, rather than only 60% damage.

File: code/modules/mob/living/carbon/superior_animal/defense.dm [hunk 1]
```diff
@@ -413,3 +413,7 @@
 	user.do_attack_animation(src)
 	spawn(1) updatehealth()
 	return TRUE
+
+/mob/living/carbon/superior_animal/getarmor(def_zone, type)
+	//log_and_message_admins("LOG 1.5: def_zone [def_zone] | type [type] | armor(type) [armor[type]].")
```
Comment (Takykardi) [inline code/modules/mob/living/carbon/superior_animal/defense.dm:418]: Please remove commented-out debug code.

File: code/modules/mob/living/living_defense.dm [hunk 2]
```diff
@@ -67,7 +67,8 @@
 
 	var/remaining_armor = armor
 	var/remaining_ablative = ablative_armor
-
+	//log_and_message_admins("LOG 1: armor [armor] | ablative_armor [ablative_armor] | remaining_armor| [remaining_armor] | remaining_ablative [remaining_ablative].")
+	//log_and_message_admins("LOG 1.2: attack_flag [attack_flag] | damagetype [damagetype] | def_zone| [def_zone] | armor_divisor [armor_divisor].")	for(var/dmg_type in dmg_types)
```
Comment (Takykardi) [inline code/modules/mob/living/living_defense.dm:71]: Ditto.

File: code/modules/mob/living/living_defense.dm [hunk 3]
```diff
@@ -152,6 +153,7 @@
 	//Feedback
 	//In order to show both target and everyone around that armor is actually working, we are going to send message for both of them
 	//Goon/tg chat should take care of spam issue on this one
+	//log_and_message_admins("LOG 2: effective_armor [effective_armor] | total_dmg [total_dmg] | dealt_damage| [dealt_damage].")
```
Comment (Takykardi) [inline code/modules/mob/living/living_defense.dm:156]: Ditto.

File: code/modules/mob/living/simple_animal/simple_animal.dm [hunk 4]
```diff
@@ -818,3 +818,7 @@
 	else
 		AI_inactive = TRUE
 		to_chat(src, SPAN_NOTICE("You toggle the mobs default AI to OFF."))
+
+/mob/living/simple_animal/getarmor(def_zone, type)
+	//log_and_message_admins("LOG 1.5: def_zone [def_zone] | type [type] | armor(type) [armor[type]].")
```
Comment (Takykardi) [inline code/modules/mob/living/simple_animal/simple_animal.dm:823]: Ditto.

File: code/modules/mob/living/carbon/superior_animal/superior_defines.dm [hunk 5]
```diff
@@ -110,7 +110,7 @@
 	/// Alpha value of our telegraph beam.
 	var/telegraph_beam_alpha = 50
 
-	var/moved = FALSE
+	// var/moved = FALSE         This appears to be unused. Commenting out for now.
```
Comment (Takykardi) [inline code/modules/mob/living/carbon/superior_animal/superior_defines.dm:113]: This seems to have been made unused due to the change seen in superior_animal/attack.dm line 12, please consider modifying comment.

**Review by Takykardi — DISMISSED** [review]

**Pull request closed without merging**